sdk-lib/src/main/java/cash/z/ecc/android/sdk/Synchronizer.kt

```diff
@@ -444,11 +444,14 @@ interface Synchronizer {
          * Synchronizer requires. It contains all information necessary to build a synchronizer and it is
          * mainly responsible for initializing the databases associated with this synchronizer and loading
          * the rust backend.
+         * @param seed the wallet's seed phrase. This only needs to be provided if this method returns an
+         * error indicating that the seed phrase is required for a database migration.
          */
         suspend fun new(
-            initializer: Initializer
+            initializer: Initializer,
+            seed: ByteArray? = null,
```

Comment:
I do not know if this is the correct way to expose this; given that `seed` is not stored in `Initializer` currently, I figured it was better to not require it there. But I have no idea how these APIs are invoked. I _think_ this is the only actually-public API related to initialization.